btk/measure.py

```diff
@@ -1,175 +1,239 @@
+                    raise ValueError(
+                        "The output catalog of at least one of your measurement functions does not"
+                        "contain the 'dx' and 'dy' columns which are mandatory for a single"
+                        "survey study."
+                    )
+
+            if isinstance(batch["blend_images"], dict):
+                if not ("ra" in out["catalog"].colnames and "dec" in out["catalog"].colnames):
+                    raise ValueError(
+                        "The output catalog of at least one of your measurement functions does not"
+                        "contain the 'ra' and 'dec' columns which are mandatory for a"
+                        "multi-resolution study."
+                    )
+
+            for key in ["deblended_images", "segmentation"]:
+                if key in out and out[key] is not None:
+                    if not isinstance(out[key], np.ndarray):
+                        raise TypeError(
+                            f"The output '{key}' of at least one of your measurement"
+                            f"functions is not a numpy array."
+                        )
+                    if not out[key].shape[-2:] == batch["blend_images"].shape[-2:]:
+                        raise ValueError(
+                            f"The shapes of the blended images in your {key} don't"
+                            f"match for at least one your measurement functions."
+                        )
+
+            out = {k: out.get(k, None) for k in self.measure_params}
+            output.append(out)
+        return output
 
     def __next__(self):
         """
         Returns:
-            draw_blend_generator output, deblender output and measurement output.
+            draw_blend_generator output from `__next__` method.
+            measurement output: List of len(batch_size), where each element is a list of
+                                len(measure_functions) corresponding to the measurements made by
+                                each function on each element of the batch. If only one
+                                measure_function was passed in, then each sub-list is converted
+                                to a single element.
         """
 
         blend_output = next(self.draw_blend_generator)
-        deblend_results = {}
-        measured_results = {}
         input_args = [(blend_output, i) for i in range(self.batch_size)]
-        batch_results = multiprocess(
+        measure_results = multiprocess(
             self.run_batch,
             input_args,
             self.cpus,
             self.multiprocessing,
             self.verbose,
         )
-        for i in range(self.batch_size):
-            deblend_results.update({i: batch_results[i][0]})
-            measured_results.update({i: batch_results[i][1]})
+        for i, r in enumerate(measure_results):
+            if len(r) == 1:
+                measure_results[i] = r[0]
```

Comment:
What is the purpose of this ?

Reply:
We have a "list in, list out" approach for measure_functions ; if you provide a single function, you want to get your results directly, not as a list with a single element.

Reply:
Then it's no more "list in, list out". And it makes subsequent tasks more difficult because you may need to handle cases where the measurement is not a list..

Reply:
I think @aboucaud is right, handling this special case might be problematic. I wonder if that's an issue with the multiple survey code as well